src/core/utils/history_tracker.cpp

```diff
@@ -425,6 +425,21 @@ void HistoryTracker::RecordExternalRouteEvent(NetDataEvent aEvent, const Network
 
 #endif // OPENTHREAD_CONFIG_HISTORY_TRACKER_NET_DATA
 
+#if OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE && OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
+void HistoryTracker::RecordEpskcEvent(MeshCoP::BorderAgent::EphemeralKeyManager::Event              aEvent,
+                                      MeshCoP::BorderAgent::EphemeralKeyManager::DeactivationReason aReason)
+{
+    EpskcInfo *entry = mEpskcInfoHistory.AddNewEntry();
+
+    VerifyOrExit(entry != nullptr);
+    entry->mEvent  = MapEnum(aEvent);
+    entry->mReason = MapEnum(aReason);
+
+exit:
+    return;
+}
+#endif
```

Comment:
We also need to add something to `HistoryTracker::HandleTimer()` like 
```c++
#if OPENTHREAD_CONFIG_BORDER_AGENT_ENABLE && OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE
   mEpskcInfoHistory.UpdateAgedEntries();
#endif
```

The `UpdateAgedEntries()` handles the proper aging of entries beyond 49 days.